src/main/java/study1/step3/CarList.java

```diff
@@ -0,0 +1,24 @@
+package study1.step3;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class CarList {
+    private final List<Car> carList = new ArrayList<>();
+    Decision moveDecisionMaker;
+
+    CarList(int carCnt, Decision moveDecisionMaker){
+        this.moveDecisionMaker = moveDecisionMaker;
+        for (int i = 0; i < carCnt; i ++){
+            carList.add(new Car());
+        }
+    }
+
+    public void race(){
+        this.carList.forEach(car -> car.move(moveDecisionMaker));
+    }
```

Comment:
```suggestion
    public void race(MoveDecisionMaker moveDecisionMaker){
        this.carList.forEach(car -> car.move(moveDecisionMaker));
    }
```
앞서 언급한 내용을 반영하면 다음과 같습니다.